types/node/stream.d.ts

```diff
@@ -901,6 +938,22 @@ declare module 'stream' {
             readable?: boolean | undefined;
             writable?: boolean | undefined;
         }
+        /**
+         * Combines two or more streams into a `Duplex` stream that writes to the
+         * first stream and reads from the last. Each provided stream is piped into
+         * the next, using `stream.pipeline`. If any of the streams error then all
+         * are destroyed, including the outer `Duplex` stream.
+         *
+         * Because `stream.compose` returns a new stream that in turn can (and
+         * should) be piped into other streams, it enables composition. In contrast,
+         * when passing streams to `stream.pipeline`, typically the first stream is
+         * a readable stream and the last a writable stream, forming a closed
+         * circuit.
+         *
+         * @since v16.9.0
+         * @param streams
+         */
+        function compose(...streams: Stream[]): Duplex;
```

Comment:
FIrst arg should be writable no?

Reply:
Would this be more like what you expect? It does not follow the docs tho.

```
function compose(writable: Writable | NodeJS.WritableStream): Duplex;
function compose(writable: Writable | NodeJS.WritableStream, ...streams: Stream[]): Duplex;
```

Reply:
Hmm, I guess that technically every stream must be a writeable stream? Since each stream will get piped to the next stream. In fact, I guess that every stream must be a duplex stream? 🤔 

We should probably look at how `pipeline` is typed, but skip the first and last argument (since `compose` has no "source" and no "destination"):

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0a931721ebb21055134dcee12927bb0f96381ffd/types/node/stream.d.ts#L1018-L1219

Reply:
So, I would rather remove the `compose` typings from this PR so that they can have their own due course discussion before landing on @types/node. It's not a feature i particularly care about, I only did this PR to get the crypto additions in but didn't want to bump the version without doing the rest of the changes.

Reply:
@LinusU i've tabled the `compose` typings for later > 527c7aa8847189fc79f62d3c83891edc94828079